### PR TITLE
Removes UI Framework KUI doc site

### LINKED ISSUE #1096: WS-2020-0208 (Medium) detected in highlight.js-9.18.5.tgz
## WS-2020-0208 - Medium Severity Vulnerability
<details><summary> Vulnerable Library - <b>highlight.js-9.18.5.tgz</b></p></summary>

<p>Syntax highlighting with language autodetection.</p>
<p>Library home page: <a href="https://registry.npmjs.org/highlight.js/-/highlight.js-9.18.5.tgz">https://registry.npmjs.org/highlight.js/-/highlight.js-9.18.5.tgz</a></p>
<p>

Dependency Hierarchy:
  - @osd/ui-framework-1.0.0.tgz (Root Library)
    - :x: **highlight.js-9.18.5.tgz** (Vulnerable Library)
<p>Found in HEAD commit: <a href="https://github.com/opensearch-project/OpenSearch-Dashboards/commit/4fd064970b66ce555f48c22dfab6ed965d0e260a">4fd064970b66ce555f48c22dfab6ed965d0e260a</a></p>
<p>Found in base branch: <b>main</b></p>
</p>
</details>
<p></p>
<details><summary> Vulnerability Details</summary>
<p>  
  
If are you are using Highlight.js to highlight user-provided data you are possibly vulnerable. On the client-side (in a browser or Electron environment) risks could include lengthy freezes or crashes... On the server-side infinite freezes could occur... effectively preventing users from accessing your app or service (ie, Denial of Service).  This is an issue with grammars shipped with the parser (and potentially 3rd party grammars also), not the parser itself. If you are using Highlight.js with any of the following grammars you are vulnerable. If you are using highlightAuto to detect the language (and have any of these grammars registered) you are vulnerable.

<p>Publish Date: 2020-12-04
<p>URL: <a href=https://github.com/highlightjs/highlight.js/commit/373b9d862401162e832ce77305e49b859e110f9c>WS-2020-0208</a></p>
</p>
</details>
<p></p>
<details><summary> CVSS 3 Score Details (<b>5.3</b>)</summary>
<p>

Base Score Metrics:
- Exploitability Metrics:
  - Attack Vector: Network
  - Attack Complexity: Low
  - Privileges Required: None
  - User Interaction: None
  - Scope: Unchanged
- Impact Metrics:
  - Confidentiality Impact: Low
  - Integrity Impact: None
  - Availability Impact: None
</p>
For more information on CVSS3 Scores, click <a href="https://www.first.org/cvss/calculator/3.0">here</a>.
</p>
</details>
<p></p>
<details><summary> Suggested Fix</summary>
<p>

<p>Type: Upgrade version</p>
<p>Origin: <a href="https://github.com/highlightjs/highlight.js/tree/10.4.1">https://github.com/highlightjs/highlight.js/tree/10.4.1</a></p>
<p>Release Date: 2020-12-04</p>
<p>Fix Resolution: 10.4.1</p>

</p>
</details>
<p></p>

### PR DESCRIPTION
**Pull request by tmarkley**
### Description
* KUI components are no longer being developed and the dependencies for the doc site are related to CVEs.
* Removes related grunt tasks.
* Removes all unused dependencies from the osd/ui-framework package.
 
### Issues Resolved
Resolves #1096
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
    - [x] `yarn test:jest`
    - [x] `yarn test:jest_integration`
    - [x] `yarn test:ftr`
- [x] New functionality has been documented.
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: packages/osd-ui-framework/Gruntfile.js [hunk 1]
```diff
@@ -73,180 +64,6 @@ module.exports = function (grunt) {
...
-
-    uiFrameworkServerBuild.then(done);
-  });
-
-  grunt.registerTask('docSiteStart', function () {
-    const done = this.async();
-    Promise.all([uiFrameworkWatch(), uiFrameworkServerStart()]).then(done);
-  });
-
-  grunt.registerTask('compileCssLight', function () {
```
Comment (ashwin-pc) [inline packages/osd-ui-framework/Gruntfile.js:115]: Should we be removing this? The compiled css files do seem to be used in [uiRenderMixin](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/legacy/ui/ui_render/ui_render_mixin.js#L119)

File: packages/osd-ui-framework/Gruntfile.js [hunk 2]
```diff
@@ -73,180 +64,6 @@ module.exports = function (grunt) {
...
-    const done = this.async();
-    uiFrameworkCompileLight().then(done);
-  });
-
-  grunt.registerTask('compileCssDark', function () {
```
Comment (ashwin-pc) [inline packages/osd-ui-framework/Gruntfile.js:120]: The same for this

**Review by ashwin-pc — COMMENTED** [review]
The rest looks good. Just a question about removing the grunt tasks to compile the scss for ui-framework

File: packages/osd-ui-framework/Gruntfile.js (hunk 1 above)
Comment (tmarkley) [reply packages/osd-ui-framework/Gruntfile.js:115]: Yes, good eye. When I first looked at this I glossed over the fact that the KUI css has to be compiled manually to the /dist directory every time there is a change. :/

Comment (tmarkley) [reply packages/osd-ui-framework/Gruntfile.js:115]: Refactored those tasks.

**Comment by tmarkley** [comment]
I successfully ran the `yarn uiFramework:compileCss` script but it removed all of the Webkit CSS extensions from the light and dark compiled CSS, so I reverted those changes:

```
$ yarn uiFramework:compileCss
yarn run v1.22.17
$ cd packages/osd-ui-framework && yarn compileCss
$ grunt compileCss
Running "compileCss" task

Done.
Done in 2.67s.
```

**Comment by ashwin-pc** [comment]
> I successfully ran the `yarn uiFramework:compileCss` script but it removed all of the Webkit CSS extensions from the light and dark compiled CSS, so I reverted those changes:

Ran that on `main`, looks like that's unrelated to your change. The rest looks good to me!

**Review by ashwin-pc — DISMISSED** [review]

**Review by boktorbb-amzn — DISMISSED** [review]

**Comment by kavilla** [comment]
Looks like this doesn't actually had an impact on the CVE and the CVE was addressed. Given that has conflicts and the CVE was addressed I will remove the `2.0.0` label.

**Comment by tmarkley** [comment]
> Looks like this doesn't actually had an impact on the CVE and the CVE was addressed. Given that has conflicts and the CVE was addressed I will remove the `2.0.0` label.

@kavilla the CVE was not addressed. It's still in the yarn.lock file: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/yarn.lock

**Comment by kavilla** [comment]
> > Looks like this doesn't actually had an impact on the CVE and the CVE was addressed. Given that has conflicts and the CVE was addressed I will remove the `2.0.0` label.
> 
> @kavilla the CVE was not addressed. `highlight.js@9.18.5` is still in the yarn.lock file: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/yarn.lock

Apologies, when I checked the CVE it was automatically resolved but I guess it re-opened 4 days ago. I see there are conflicts here.

**Review by boktorbb-amzn — APPROVED** [review]

**Comment by kavilla** [comment]
Similar with the other PR with the lock file being changed do we see any potential of breakage with plugins? Grant it is the lock file although we do is for handling a lot of our resolutions.

Thanks!

**Comment by tmarkley** [comment]
> Similar with the other PR with the lock file being changed do we see any potential of breakage with plugins? Grant it is the lock file although we do is for handling a lot of our resolutions.
> 
> Thanks!

Yeah I think there could be some risk here with the amount of dependencies that we're removing. I'm okay to hold off until next week and merge after RC1.

**Review by kavilla — CHANGES_REQUESTED** [review]
Blocking for rc1 release.

**Review by ashwin-pc — APPROVED** [review]

**Review by boktorbb-amzn — APPROVED** [review]

**Pull request merged**